### DISCUSSION TITLE
A JSON Viewer - my upcoming project

### DISCUSSION THREAD

**Discussion by rexkogitans** [romgrk/node-gtk#315] [Show and tell]
I have got an idea for a small software project, a JSON viewer, which started out in Python (my first touch with Pyhton) with GTK 3. Midway, I found out that GTK 4 arrived and I could not make friends with Python. My project was halted for several months.
After tinkering about 3 months with various programming languages (C, Ruby, Node-JS) and Toolkits (GTK, QT, tk), I finally found my decision in - drum roll, please - Node-JS with GTK. The GUI will be designed with Cambalache.

Although not being able to bind the signals automatically, I decided that this is the least drawback since the app won't have too many signals. It won't make an unmaintable list in the classes.

Furthermore, I think Node-GTK should receive more attention as it is a great project. I am looking forward to see it advance as the number of projects depending on it grows.

I will keep you up to date as soon as my project is in a state where I will push it to a public Git repository for the first time.

**Comment by gavr123456789** [comment]
But after all, Cambalache is not ready yet and is at a very early stage of development. But after all, Cambalache is not ready yet and is at a very early stage of development. By the way, I also plan to make a json viewer, but embedded in my file browser(Im using nim now, waiting for typescript support).

**Reply by rexkogitans** [reply]
That's interesting. In my former company, I wrote a JSON viewer in pure HTML/JS/CSS in a single file. AFAIR, also the icons I used were hard coded as Base64-Strings.
Yes, Cambalache is in a very early stage. It happend that Juan Pablo Ugarte fixed a bug I reported him, so maybe my project will also help his work.
